src/backend/env/mod.rs

```diff
@@ -1698,11 +1698,16 @@ impl State {
                 state.mint(market_price);
                 state.minting_mode = false;
             });
-        }
+            revenue
+        } else {
+            0
+        };
 
         State::distribute_icp().await;
 
         mutate(|state| {
+            state.distribute_revenue_from_icp(auction_revenue);
```

Comment:
I don't see the difference from before except that it is locking together multiple updates